src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip

```diff
@@ -0,0 +1,229 @@
+@extends "csharp/common.snip"
+
+@snippet generate(common)
+    // Copyright 2016 Google Inc. All Rights Reserved.
+    //
+    // Licensed under the Apache License, Version 2.0 (the "License");
+    // you may not use this file except in compliance with the License.
+    // You may obtain a copy of the License at
+    //
+    //     http://www.apache.org/licenses/LICENSE-2.0
+    //
+    // Unless required by applicable law or agreed to in writing, software
+    // distributed under the License is distributed on an "AS IS" BASIS,
+    // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    // See the License for the specific language governing permissions and
+    // limitations under the License.
+
+    // Generated code. DO NOT EDIT!
+
+    @join import : common.api.imports
```

Comment:
I would much prefer this snippets file to receive a different object from the one that the main surface file uses. This snippet file uses a very small subset of the data, and thus it is a little confusing that it is used here. You might want something like SnippetsFileView or something like that. You could then put in exactly the api methods that you want to generate snippets for, and you wouldn't need the if condition below to skip FlattenedAsyncCancellationTokenMethod.